Engine/src/main/java/nullengine/item/ItemImpl.java

```diff
@@ -0,0 +1,129 @@
+package nullengine.item;
+
+import nullengine.block.Block;
+import nullengine.component.Component;
+import nullengine.entity.Entity;
+import nullengine.event.Event;
+import nullengine.event.world.item.ItemBreakEvent;
+import nullengine.event.world.item.ItemThrowEvent;
+import nullengine.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public class ItemImpl implements Item,Runnable {
```

Comment:
I'm just curious. Why ItemImpl is a Runnable?

What is the design here? Am I missing any context about this? @Mouse0w0